Sources/Common/Background Queue/QueueQueryRunner.swift

```diff
@@ -15,18 +16,23 @@ internal class CioQueueQueryRunner: QueueQueryRunner {
         self.logger = logger
     }
 
-    func getNextTask(_ queue: [QueueTaskMetadata], lastFailedTask: QueueTaskMetadata?) -> QueueTaskMetadata? {
-        guard !queue.isEmpty else {
+    func getNextTask(
+        _ inventory: [QueueTaskMetadata],
+        lastRanTask: QueueTaskMetadata?,
+        lastFailedTask: QueueTaskMetadata?
+    ) -> QueueTaskMetadata? {
+        guard !inventory.isEmpty else {
             return nil
         }
+        let inventory = getInventoryTasksLeftToRun(inventory: inventory, lastRanTask: lastRanTask)
         if let lastFailedTask = lastFailedTask {
             updateCriteria(lastFailedTask: lastFailedTask)
         }
 
         // log *after* updating the criteria
         logger.debug("queue querying next task. criteria: \(queryCriteria)")
 
-        return queue.first(where: { doesTaskPassCriteria($0) })
+        return inventory.first(where: { doesTaskPassCriteria($0) })
```

Comment:
Out of curiosity, what is difference between queue and inventory ?

Reply:
The queue consists of: [an inventory of tasks](https://github.com/customerio/customerio-ios/blob/606edb3b62dc38e94667e6177bfef470ffbf6e6f/Sources/Common/Background%20Queue/QueueStorage.swift#L27) and [a list of tasks](https://github.com/customerio/customerio-ios/blob/606edb3b62dc38e94667e6177bfef470ffbf6e6f/Sources/Common/Background%20Queue/QueueStorage.swift#L33). 2 different data types for memory efficiency. When I talk about the queue, I often refer to "the inventory of tasks" in the queue. 

I made this variable name change because I felt it made more sense for the change that I made. I find this change very minor and an open to ideas if someone has a different idea.